core/esmf-aspect-meta-model-java/src/main/java/org/eclipse/esmf/aspectmodel/loader/AspectModelLoader.java

```diff
@@ -173,6 +178,37 @@ public AspectModel load( final InputStream inputStream ) {
       return buildAspectModel( loaderContext.loadedFiles() );
    }
 
+   public AspectModel loadFromArchive( final String pathToArchive ) {
+      File zipFile = new File( pathToArchive );
+
+      if ( !zipFile.exists() || !zipFile.isFile() ) {
+         throw new RuntimeException( new FileNotFoundException( "The specified file does not exist or is not a file." ) );
+      }
+
+      List<AspectModelFile> aspectModelFiles = new ArrayList<>();
+
+      try ( ZipFile zip = new ZipFile( zipFile ) ) {
+         Enumeration<? extends ZipEntry> entries = zip.entries();
+
+         while ( entries.hasMoreElements() ) {
+            ZipEntry entry = entries.nextElement();
+
+            if ( entry.getName().endsWith( ".ttl" ) ) {
+               try ( InputStream inputStream = zip.getInputStream( entry ) ) {
+                  AspectModelFile aspectModelFile = AspectModelFileLoader.load( inputStream );
+                  aspectModelFiles.add( aspectModelFile );
```

Comment:
At this point, `migrate()` needs to be called for the loaded `AspectModelFile` in order to do the automatic migration from previous meta model levels.